### PR TITLE
Add genAI endpoint for explaining why user's constructed response item (CRI) is incorrect 

### PR DESCRIPTION
**Pull request by jchoi92k**
- API endpoint expects student's response and the page_slug and chunk_slug. 

```
page_slug: str
chunk_slug: str
student_response: str
```

- Uses existing method (strapi.get_chunk(page_slug, chunk_slug)) to get relevant data from our Strapi DB.
- Includes version control for scipy (scipy==1.12) because of an issue with the newest stable version of scipy (1.13) [https://stackoverflow.com/questions/78279136/importerror-cannot-import-name-triu-from-scipy-linalg-gensim](https://stackoverflow.com/questions/78279136/importerror-cannot-import-name-triu-from-scipy-linalg-gensim)

- Flagging that two tests from the main branch (`test_summary_eval.py` and `test_summary_eval_stairs`) are not working (when tested directly from the main branch without any of my changes). Not sure what's going on here - maybe it's related to my dev container setup? In any case, doesn't seem related to any of the changes I've made in this branch since the test fails for the main branch in my setup.
- Let me know if anything needs to be changed.

### CODE REVIEW

File: src/pipelines/chat.py
Comment (langdonholmes) [inline src/pipelines/chat.py:1]: I'd prefer to use the existing chat pipeline for this. We can add the "Your response" preface as a postprocessing step in the main endpoint definition.

**Review by langdonholmes — COMMENTED** [review]
Hi Joon, could you use the existing chat_pipeline for this? Staying DRY for maintainability. 

You can do one of the following:
1. Define a new generator in the `src.main` endpoint definition. Inside this generator, preface each response with "Your response".
2. Modify the existing chat_pipeline to allow for an optional `preface_text` keyword argument and use this to modify each `request_output.outputs[0].text`

**Comment by langdonholmes** [comment]
Failed tests are caused by the page `what-is-law` missing from the vector-store. I will change the tests to use a different page from an active volume.

**Comment by jchoi92k** [comment]
@langdonholmes Makes sense! Changes are in place. 
Also changed the summary tests' page slugs and now all tests are working.

**Review by langdonholmes — APPROVED** [review]
Thanks for your work here, Joon!

**Pull request merged**